From 69cb20fb25ed0fa7c2bcaec62fc03c9cfb3944c9 Mon Sep 17 00:00:00 2001
From: Yongzhi Chen <ychena@apache.org>
Date: Fri, 30 Sep 2016 10:39:11 -0400
Subject: [PATCH 676/723] CDH-44088 CDH-45433 HIVE-14784: Operation logs are
 disabled automatically if the parent directory does
 not exist. (Naveen Gangam via Yongzhi Chen)

Change-Id: I8f43f2519df2b307c528a0a4bdcd67d92d481a66
---
 .../hive/service/cli/operation/Operation.java      |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/service/src/java/org/apache/hive/service/cli/operation/Operation.java b/service/src/java/org/apache/hive/service/cli/operation/Operation.java
index 8767f7e..da2ff40 100644
--- a/service/src/java/org/apache/hive/service/cli/operation/Operation.java
+++ b/service/src/java/org/apache/hive/service/cli/operation/Operation.java
@@ -220,6 +220,17 @@ protected void createOperationLog() {
               operationLogFile.getAbsolutePath());
           operationLogFile.delete();
         }
+        if (!operationLogFile.getParentFile().exists()) {
+          LOG.warn("Operations log directory for this session does not exist, it could have been deleted " +
+              "externally. Recreating the directory for future queries in this session but the older operation " +
+              "logs for this session are no longer available");
+          if (!operationLogFile.getParentFile().mkdir()) {
+            LOG.warn("Log directory for this session could not be created, disabling " +
+                "operation logs: " + operationLogFile.getParentFile().getAbsolutePath());
+            isOperationLogEnabled = false;
+            return;
+          }
+        }
         if (!operationLogFile.createNewFile()) {
           // the log file already exists and cannot be deleted.
           // If it can be read/written, keep its contents and use it.
-- 
1.7.9.5

